cdisc_rules_engine/utilities/report_factory.py

```diff
@@ -0,0 +1,58 @@
+from typing import List
+from cdisc_rules_engine.enums.report_types import ReportTypes
+from cdisc_rules_engine.models.rule_validation_result import RuleValidationResult
+from cdisc_rules_engine.models.validation_args import Validation_args
+from cdisc_rules_engine.utilities.excel_report import ExcelReport
+from cdisc_rules_engine.utilities.json_report import JsonReport
+from cdisc_rules_engine.services.data_services.local_data_service import (
+    LocalDataService,
+)
+
+
+class ReportFactory:
+    """
+    This class is a factory that creates reporting service depending on the output format.
+
+    Constructor arguments:
+    data_path -- path to the output file
+    results -- list with rule results generated by the engine
+    elapsed_time -- time spent on validation
+    args -- CLI arguments for the validate command
+    data_service -- instance of engine data service to read file contents
+    """
+
+    def __init__(
+        self,
+        data_path: str,
+        results: List[RuleValidationResult],
+        elapsed_time: float,
+        args: Validation_args,
+        data_service: LocalDataService,
+    ):
+        self._data_path = data_path
+        self._results = results
+        self._elapsed_time = elapsed_time
+        self._args = args
+        self._data_service = data_service
+
+    def get_report_service(self):
+        output_type = self._args.output_format.upper()
+        if output_type == ReportTypes.XLSX.value:
+            if self._data_service != None:
```

Comment:
self._data_service can't be None since it is a mandatory argument in the class init. We can remove this condition safely.

Reply:
It is done for backward compatibility with tests, which provide None as a data_service value. I understand, that probably it will be better to rewrite test, rather than having this condition.

Reply:
Yes, or in test you just can pass a mock

Reply:
Updated the test and remove the check for None from the report_factory.